test/test_analysis_endpoint.py

```diff
@@ -0,0 +1,34 @@
+import json
+import pytest
+import requests
+
+def read_payload_from_file(filename):
+    with open(filename) as file:
+        payload = json.load(file)
+    return payload
+
+@pytest.fixture(params=['data_recent.json', 'data_future.json'])
+def payload(request):
+    return read_payload_from_file(request.param)
+
+def test_analysis_endpoint(payload):
+    url = 'http://localhost:5020/api/v1/analysis'
```

Comment:
https://github.com/resource-watch/adapter-earth-engine/blob/9010bf58d48b1873b12ecf9eef0944108abe5131/tests/delete_dataset_tests.py

This is how you can test a Flask app without requiring the server to be up, and without hardcoding URLs